### PR TITLE
[SYCL] Fix kernel::get_native

### PR DESCRIPTION
**Pull request by alexbatashev**
This patch fixes compilation errors with kernel::get_native usage and improves test coverage for this method

### CODE REVIEW

File: sycl/include/CL/sycl/backend/level_zero.hpp [hunk 1]
```diff
@@ -57,7 +57,7 @@ struct interop<backend::level_zero,
 };
 
 namespace detail {
-template <> class BackendReturn<backend::level_zero, kernel> {
+template <> struct BackendReturn<backend::level_zero, kernel> {
```
Comment (s-kanaev) [inline sycl/include/CL/sycl/backend/level_zero.hpp:60]: Is it alignment with template declaration?

File: sycl/test/on-device/basic_tests/interop/get_native_ze.cpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
+// REQUIRES: level_zero, level_zero_dev_kit
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %level_zero_options %s -o %t.ze.out
+// RUN: env SYCL_DEVICE_FILTER="level_zero" %t.ze.out
```
Comment (s-kanaev) [inline sycl/test/on-device/basic_tests/interop/get_native_ze.cpp:3]: Should there also be OpenCL version of the test?

**Review by s-kanaev — DISMISSED** [review]
Looks good overall.
@alexbatashev , could you, please, provide a brief description of the problem being fixed in patch description?

File: sycl/test/on-device/basic_tests/interop/get_native_ze.cpp (hunk 2 above)
Comment (bader) [inline sycl/test/on-device/basic_tests/interop/get_native_ze.cpp:3]: I think `// REQUIRES: level_zero` configures `SYCL_DEVICE_FILTER` environment variable.
Do we really need to reset it in the test?

File: sycl/include/CL/sycl/backend/level_zero.hpp (hunk 1 above)
Comment (alexbatashev) [reply sycl/include/CL/sycl/backend/level_zero.hpp:60]: Yes

File: sycl/test/on-device/basic_tests/interop/get_native_ze.cpp (hunk 2 above)
Comment (alexbatashev) [reply sycl/test/on-device/basic_tests/interop/get_native_ze.cpp]: Sure, why not

Comment (alexbatashev) [reply sycl/test/on-device/basic_tests/interop/get_native_ze.cpp]: Fixed

**Review by s-kanaev — APPROVED** [review]

**Pull request merged**